code/modules/hydroponics/gene_modder.dm

```diff
@@ -0,0 +1,398 @@
+/obj/machinery/plantgenes
+	name = "plant DNA manipulator"
+	desc = "An advanced device designed to manipulate plant genetic makeup."
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "dnamod"
+	density = 1
+	anchored = 1
+
+	var/obj/item/seeds/seed
+	var/obj/item/weapon/disk/plantgene/disk
+
+	var/list/core_genes = list()
+	var/list/reagent_genes = list()
+	var/list/trait_genes = list()
+
+	var/datum/plant_gene/target
+	var/operation = ""
+	var/rating = 0 // TODO: add failures, make it useful
+
+/obj/machinery/plantgenes/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/weapon/circuitboard/plantgenes(src)
+	component_parts += new /obj/item/weapon/stock_parts/console_screen(src)
+	component_parts += new /obj/item/weapon/stock_parts/scanning_module(src)
+	component_parts += new /obj/item/weapon/stock_parts/micro_laser(src)
+	component_parts += new /obj/item/weapon/stock_parts/micro_laser(src)
+	component_parts += new /obj/item/weapon/stock_parts/manipulator(src)
+	RefreshParts()
+
+/obj/machinery/plantgenes/RefreshParts()
+	rating = 0
+	for(var/obj/item/weapon/stock_parts/S in component_parts)
+		rating += S.rating-1
+
+/obj/machinery/plantgenes/update_icon()
+	..()
+	overlays.Cut()
+	if((stat & (BROKEN|NOPOWER)))
+		icon_state = "dnamod-off"
+	else
+		icon_state = "dnamod"
+	if(seed)
+		overlays += "dnamod-dna"
+	if(panel_open)
+		overlays += "dnamod-open"
+
+/obj/machinery/plantgenes/attackby(obj/item/I, mob/user, params)
+	if(default_deconstruction_screwdriver(user, "dnamod", "dnamod", I))
```

Comment:
It can't handle my overlays. And it uses overlays because it has a depowered icon_state.